### PR TITLE
Remove hack for loading metadata exchange config

### PR DESCRIPTION
**Pull request by gargnupur**
Ref: https://github.com/istio/istio/issues/20395

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[X ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: tests/integration/telemetry/stats/prometheus/tcp/testdata/metadata_exchange_tcp_filter.yaml [hunk 1]
```diff
@@ -32,8 +32,10 @@ spec:
           filters:
           - name: envoy.filters.network.upstream.metadata_exchange
             typed_config:
-              "@type": type.googleapis.com/envoy.tcp.metadataexchange.config.MetadataExchange
-              protocol: istio-peer-exchange
+              "@type": type.googleapis.com/udpa.type.v1.TypedStruct
+              type_url: type.googleapis.com/envoy.tcp.metadataexchange.config.MetadataExchange
+              value:
+                protocol: istio-peer-exchange
```
Comment (howardjohn) [inline tests/integration/telemetry/stats/prometheus/tcp/testdata/metadata_exchange_tcp_filter.yaml]: what about the install charts?

**Review by howardjohn — APPROVED** [review]

**Comment by mandarjog** [comment]
Everything failing with 
‘’’ go: github.com/istio.io/proxy/src/envoy/tcp/metadata_exchange/config@v0.0.0: reading https://proxy.golang.org/github.com/istio.io/proxy/src/envoy/tcp/metadata_exchange/config/@v/v0.0.0.mod: 410 Gone ‘’’

**Comment by howardjohn** [comment]
Why do we have the `github.com/istio.io/proxy/src/envoy/tcp/metadata_exchange/config` import still?

**Comment by mandarjog** [comment]
Leftover somewhere, needs to be removed.

**Comment by gargnupur** [comment]
> Leftover somewhere, needs to be removed.

was left in go.mod file require clause.. removed now.

File: tests/integration/telemetry/stats/prometheus/tcp/testdata/metadata_exchange_tcp_filter.yaml (hunk 1 above)
Comment (gargnupur) [reply tests/integration/telemetry/stats/prometheus/tcp/testdata/metadata_exchange_tcp_filter.yaml]: added

**Comment by howardjohn** [comment]
@gargnupur lgtm but some failing tests

**Review by mandarjog — COMMENTED** [review]
Lgtm

**Comment by gargnupur** [comment]
@mandarjog : Everything is passing now.. can I please get an approval from you?

**Review by mandarjog — APPROVED** [review]
/lgtm

**Comment by howardjohn** [comment]
@gargnupur youll need to manually cherrypick

**Pull request merged**